New image base for custom runner

mpinter opened the pull request:
- [x] will rework nicer if this works - context is we're using the new gha controller so it seems we need to use appropriate containers for it to work - https://github.com/actions/actions-runner-controller/issues/3634
- [x] I'll do the tags properly in followup (if it works), but they work differently on ghcr

- [x] changed to fixed version, changed the way this is updated

mpinter committed:
New image base for custom runner

mpinter committed:
Install build-essential

vidriduch commented on Dockerfile:
```diff
@@ -1,26 +1,28 @@
-FROM summerwind/actions-runner-dind:ubuntu-20.04 as prod
+FROM ghcr.io/actions/actions-runner:latest AS prod
```
I would strongly prefer to fix the version. This `:latest` has bitten me too many times. This can break things in a such a miraculous ways :smile: that you will spend a day fixing it. We can keep the bugfix version open if you prefer it

mpinter committed:
Fix runner version and add github workflow.

- fixed version in dockerfile
- build no longer runs on cronjob
- added check-runner-updates job that checks for updates and notifies

mpinter commented:
@vidriduch tested the PR creation here https://github.com/bratislava/github-runner/pull/8

vidriduch reviewed (approved):
This is nice. Good job 👍 :shipit:
